npm-publish-branch-preview/entrypoint.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+set -e
+
+RED='\033[1;31m'
+YELLOW='\033[1;33m'
+NC='\033[0m'
+
+if [ "${#NPM_AUTH_TOKEN}" -eq "0" ]
+  then
+    echo -e "${RED}ERROR: NPM_AUTH_TOKEN not detected. Please add your NPM Token to your repository's secrets.${NC}"
+    echo -e "${YELLOW}Publishing preview will not work if the pull request was created from a forked repository unless you create the pull request against your own repository.${NC}"
+    exit 1
+elif [[ "$GITHUB_HEAD_REF" = "" ]]
+  then
+    echo -e "${RED}ERROR: We suspect this workflow was not triggered from a pull request.${NC}"
+    exit 1
+elif [[ "$GITHUB_HEAD_REF" = "latest" ]]
+  then
+    echo -e "${RED}ERROR: Unable to publish preview because your branch conflicts with NPM's protected 'latest' tag.${NC}"
+    echo -e "${YELLOW}Please change the name of your branch and resubmit the pull request.${NC}"
+    exit 1
+else
+  echo "//registry.npmjs.org/:_authToken=$NPM_AUTH_TOKEN" > ~/.npmrc
+  npm publish --access=public --tag $GITHUB_HEAD_REF
```

Comment:
@cowboyd This is what they say in the [docs](https://docs.npmjs.com/cli/dist-tag):

> The simplest way to avoid semver problems with tags is to use tags that do not begin with a number or the letter v.

But contrary to the docs, I was able to confirm from my [tests](https://www.npmjs.com/package/@minkimcello/microstates/v/0.16.4-0a6a0c5) on @minkimcello/microstates that these work fine:
- v0.0a (*only two numbers with v in front*)
- a0.0.0 (*non-v and full semver*)
- release-0.0.0 (*semever at the end*)

And the tags that fail are:
- v0.0.0*
- 0.0.0*

We could escape cases where they match the two tag formats that are not accepted. Or should we just let NPM return an error?

Reply:
Yeah, let's let keep it super simple and let NPM return the error, and just report that it's an invalid tag name in the comment.

Reply:
In which comment?

Reply:
> and just report that it's an invalid tag name in the comment.

Did you mean let NPM report the error in their own error message? Or did you want me to code in a new conditional?

Reply:
If we get an error from NPM, we just let it fail. The main thing we want to make sure is that things like `cl/branch-name` will work.

Reply:
nope, this should be good.